### PR TITLE
fix: get wallet from current persona

### PR DESCRIPTION
**Pull request by lelenei**
Closes mf-1554

### CODE REVIEW

File: packages/mask/src/plugins/Avatar/hooks/usePersonas.ts [hunk 1]
```diff
@@ -1,39 +1,30 @@
...
-
         const isOwner = userId
             ? currentIdentifier?.identifier?.toText() === userId.toLowerCase()
             : currentIdentifier?.identifier &&
               identifier.identifier &&
               currentIdentifier?.identifier.toText() === identifier.identifier.toText()
-        const wallets = currentPersonaBinding?.proofs.filter((proof) => proof.platform === NextIDPlatform.Ethereum)
-        return { wallets, isOwner, binds: currentPersonaBinding, status: personaConnectStatus }
+
+        const wallets = personaBindings?.proofs.filter((proof) => proof.platform === NextIDPlatform.Ethereum) ?? []
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/hooks/usePersonas.ts]:
```suggestion
        const wallets = personaBindings?.proofs.filter((proof) => proof.platform === NextIDPlatform.Ethereum) ?? EMPTY_LIST
```

File: packages/mask/src/plugins/Avatar/hooks/usePersonas.ts [hunk 2]
```diff
@@ -1,39 +1,30 @@
...
     const platform = activatedSocialNetworkUI.configuration.nextIDConfig?.platform as NextIDPlatform
     const identifier = useSubscription(context.lastRecognizedProfile)
+    const currentIdentifier = useSubscription(context.currentVisitingProfile)
     return useAsyncRetry(async () => {
-        if (!identifier?.identifier?.userId) return
-        const personaBindings = await NextIDProof.queryAllExistedBindingsByPlatform(
-            platform,
-            userId ?? identifier.identifier.userId,
-        )
+        if (!identifier?.identifier?.userId || !personaConnectStatus.currentPersona?.identifier.publicKeyAsHex) return
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/hooks/usePersonas.ts]: Can we use `currentIdentifier.publicKey`?

File: packages/mask/src/plugins/Avatar/hooks/usePersonas.ts [hunk 3]
```diff
@@ -1,39 +1,30 @@
...
         const isOwner = userId
             ? currentIdentifier?.identifier?.toText() === userId.toLowerCase()
             : currentIdentifier?.identifier &&
               identifier.identifier &&
               currentIdentifier?.identifier.toText() === identifier.identifier.toText()
-        const wallets = currentPersonaBinding?.proofs.filter((proof) => proof.platform === NextIDPlatform.Ethereum)
-        return { wallets, isOwner, binds: currentPersonaBinding, status: personaConnectStatus }
+
+        const wallets = personaBindings?.proofs.filter((proof) => proof.platform === NextIDPlatform.Ethereum) ?? []
+        return { wallets, isOwner, binds: personaBindings, status: personaConnectStatus }
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/hooks/usePersonas.ts]: The name of this hook confused me cause it didn't return a list of personas as its name said. Any proper name we could use instead?

File: packages/mask/src/plugins/Avatar/hooks/usePersonas.ts (hunk 2 above)
Comment (lelenei) [reply packages/mask/src/plugins/Avatar/hooks/usePersonas.ts]: currentIdentifier  has no attribute publicKey

File: packages/mask/src/plugins/Avatar/hooks/useNextIDPersonaAndWalletsByUserId.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { EMPTY_LIST, NextIDPlatform } from '@masknet/shared-base'
+import { NextIDProof } from '@masknet/web3-providers'
+import { useAsyncRetry } from 'react-use'
+import { useSubscription } from 'use-subscription'
+import { useCurrentPersonaConnectStatus } from '../../../components/DataSource/usePersonaConnectStatus'
+import { activatedSocialNetworkUI } from '../../../social-network'
+import { context } from '../context'
+
+export function useNextIDPersonaAndWalletsByUserId(userId?: string) {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/hooks/useNextIDPersonaAndWalletsByUserId.ts]: Please try `useLastRecognizedSocialIdentity` first.

File: packages/mask/src/components/DataSource/useActivatedUI.ts [hunk 5]
```diff
@@ -144,6 +144,7 @@ export function useCurrentVisitingSocialIdentity() {
             isOwner: isOwnerIdentity,
             publicKey: isOwnerIdentity ? persona?.identifier.publicKeyAsHex : first(sortedBindings)?.persona,
             hasBinding: !!bindings?.find((x) => x.persona === persona?.identifier.publicKeyAsHex.toLowerCase()),
+            binding: first(bindings?.filter((x) => x.persona === persona?.identifier.publicKeyAsHex.toLowerCase())),
```
Comment (guanbinrui) [inline packages/mask/src/components/DataSource/useActivatedUI.ts]: Avoid doing the same thing twice.

File: packages/web3-shared/base/src/specs/index.ts [hunk 6]
```diff
@@ -134,6 +134,8 @@ export interface SocialIdentity {
     publicKey?: string
     /** Is own user account identity */
     isOwner?: boolean
+
+    binding?: NextIDPersonaBindings
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/specs/index.ts:138]: Add comment.

File: packages/web3-shared/base/src/specs/index.ts [hunk 7]
```diff
@@ -134,6 +134,8 @@ export interface SocialIdentity {
     publicKey?: string
     /** Is own user account identity */
     isOwner?: boolean
+    /** persona of the social account **/
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/specs/index.ts]:
```suggestion
    /** All bindings of the persona  **/
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**